src/aio/tokio.rs

```diff
@@ -117,6 +127,35 @@ impl RedisRuntime for Tokio {
             .map(Tokio::TcpTls)?)
     }
 
+    #[cfg(all(feature = "rustls", not(feature = "tls")))]
+    async fn connect_tcp_tls(
+        hostname: &str,
+        socket_addr: SocketAddr,
+        insecure: bool,
+    ) -> RedisResult<Self> {
+        let config = create_rustls_config(insecure)?;
+        let tls_connector = TlsConnector::from(Arc::new(config));
+
+        Ok(tls_connector
+            .connect(
+                hostname.try_into()?,
+                TcpStreamTokio::connect(&socket_addr).await?,
+            )
+            .await
+            .map(Tokio::TcpTls)?)
+    }
+    #[cfg(all(feature = "rustls", feature = "tls"))]
+    async fn connect_tcp_tls(
+        _hostname: &str,
+        _socket_addr: SocketAddr,
+        _insecure: bool,
+    ) -> RedisResult<Self> {
+        fail!((
```

Comment:
This kind of thing is usually not a good idea. I think we should just pick a preferred implementation and use that one, or make the caller pick one explicitly.